core/src/main/java/io/grpc/transport/Http2ClientStream.java

```diff
@@ -136,8 +136,7 @@ protected void transportDataReceived(Buffer frame, boolean endOfStream) {
       if (endOfStream) {
         // This is a protocol violation as we expect to receive trailers.
         transportError = Status.INTERNAL.withDescription("Recevied EOS on DATA frame");
-        frame.close();
-        inboundTransportError(transportError);
+        inboundTransportError(transportError, true);
```

Comment:
Wait... We shouldn't need to request another message. Something else isn't right.